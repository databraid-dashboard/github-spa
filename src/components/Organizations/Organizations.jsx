/* eslint-disable import/no-named-as-default, consistent-return */
import React, { Component } from 'react';
import { Menu, Header, Icon, Grid, Button } from 'semantic-ui-react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { renderLogin } from '../../actions/renderActions';
import { retrieveOrgs } from '../../actions/orgActions';
import Org from '../Org/Org';
import Logout from '../Logout/Logout';
import injectWidgetId from '../../utils/utils';
import './Organizations.css';

function orgComponents(orgIds) {
  return orgIds.map(id => <Org key={id} orgId={id} />);
}

export class Organizations extends Component {
  componentDidMount() {
    this.props.retrieveOrgs(this.props.userName);
  }

  render() {
    return (
      <div>
        <Menu compact>
          <Menu.Item>
            <Button icon onClick={() => this.props.renderLogin()}>
              <Icon name="arrow left" />
            </Button>
          </Menu.Item>
          <Menu.Item>
            <Logout />
          </Menu.Item>
        </Menu>
        <Grid centered columns={3} padded>
          <Header as="h2" icon textAlign="center">
            <Icon name="github" />
            <Header.Content>Which organization are you interested in?</Header.Content>
          </Header>
          <Grid.Row>
            {orgComponents(this.props.orgIds)}
          </Grid.Row>
        </Grid>
      </div>
    );
  }
}

Organizations.propTypes = {
  retrieveOrgs: PropTypes.func.isRequired,
  orgIds: PropTypes.arrayOf(PropTypes.string).isRequired,
  renderLogin: PropTypes.func.isRequired,
  userName: PropTypes.string.isRequired,
};

Organizations.defaultProps = {
  orgIds: [],
  userName: '',
};

export const mapStateToProps = (state, ownProps) => {
  const id = ownProps.widgetId;
  const orgIds = state.widgets.byId[id].orgs.ids;
  const orgsById = state.widgets.byId[id].orgs.orgsById;
  const userName = state.widgets.byId[id].currentPage.userName;

  return {
    orgIds,
    orgsById,
    userName,
  };
};

export const mapDispatchToProps = dispatch =>
  bindActionCreators({
    retrieveOrgs,
    renderLogin,
  }, dispatch);

export default injectWidgetId(connect(mapStateToProps, mapDispatchToProps)(Organizations));
