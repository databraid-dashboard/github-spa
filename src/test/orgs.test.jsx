import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import { Organizations, mapStateToProps, mapDispatchToProps } from '../components/Organizations/Organizations';

const state = {
  widgets: {
    byId: {
      github: {
        orgs: {
          ids: [246869449],
          orgsById: {
            246869449: {
              avatarUrl: 'https://avatars1.githubusercontent.com/u/29614654?v=4',
              orgName: 'ShareCastG52',
              reposUrl: 'https://api.github.com/orgs/ShareCastG52/repos',
            },
          },
        },
        currentPage: { render: 'orgs', repoName: null, userName: 'michaelmurray6298' },
      },
    },
  },
};

describe('Organizations component', () => {
  it('should render a component with props as specified ', () => {
    const retrieveOrgs = jest.fn();
    const renderLogin = jest.fn();
    const component = shallow(
      <Organizations
        orgIds={state.widgets.byId.github.orgs.ids}
        loadingOrganizations={false}
        currentPage={state.widgets.byId.github.currentPage.render}
        retrieveOrgs={retrieveOrgs}
        renderLogin={renderLogin}
        userName={state.widgets.byId.github.currentPage.userName}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });
  // NOTE this test will be included in future versions
  // to check that passed in functions have been called
  // it('Should call the retrieveOrganizations Function', ()=> {
  //   const retrieveOrgs = jest.fn();
  //   const store = mockStore({ })
  //   const component = mount (
  //     <Organizations orgIds={state.orgs.ids}  retrieveOrgs={retrieveOrgs}/>,
  //   );
  //   expect(retrieveOrgs).toBeCalled();
  // });

  it('map\'s given State To Props ', () => {
    const expected = {
      orgIds: [246869449],
      orgsById: {
        246869449: {
          avatarUrl: 'https://avatars1.githubusercontent.com/u/29614654?v=4',
          orgName: 'ShareCastG52',
          reposUrl: 'https://api.github.com/orgs/ShareCastG52/repos',
        },
      },
      userName: 'michaelmurray6298',
    };
    expect(mapStateToProps(state, { widgetId: 'github' })).toEqual(expected);
  });

  it('maps component dispatches to props', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch)).toHaveProperty('retrieveOrgs');
  });
});
