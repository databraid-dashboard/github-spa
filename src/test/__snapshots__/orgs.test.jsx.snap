// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organizations component should render a component with props as specified  1`] = `
<div>
  <Menu
    compact={true}
  >
    <MenuItem>
      <Button
        as="button"
        icon={true}
        onClick={[Function]}
      >
        <Icon
          as="i"
          name="arrow left"
        />
      </Button>
    </MenuItem>
    <MenuItem>
      <Component />
    </MenuItem>
  </Menu>
  <Grid
    centered={true}
    columns={3}
    padded={true}
  >
    <Header
      as="h2"
      icon={true}
      textAlign="center"
    >
      <Icon
        as="i"
        name="github"
      />
      <HeaderContent>
        Which organization are you interested in?
      </HeaderContent>
    </Header>
    <GridRow>
      <Wrapper
        orgId="246869449"
      />
    </GridRow>
  </Grid>
</div>
`;
